packages/primitives/tooltip/src/tooltip-root.directive.ts

```diff
@@ -0,0 +1,285 @@
+            return;
+        }
+
+        this.overlayRef
+            .outsidePointerEvents()
+            .pipe(takeUntilDestroyed(this.destroyRef))
+            .subscribe((event) => this.tooltipContentDirective().onPointerDownOutside.emit(event));
+    }
+
+    private handleDelayedOpen(): void {
+        window.clearTimeout(this.openTimer);
+
+        this.openTimer = window.setTimeout(() => {
+            this.wasOpenDelayed.set(true);
+            this.setOpen(true);
+        }, this.delayDuration());
+    }
+
+    private setOpen(open = false): void {
+        if (open) {
+            this.onOpen();
+
+            document.dispatchEvent(new CustomEvent('tooltip.open'));
+        } else {
+            this.onClose();
+        }
+
+        this.isOpen.set(open);
+        this.onOpenChange.emit(open);
+    }
+
+    private createOverlayRef(): OverlayRef {
+        if (this.overlayRef) {
+            return this.overlayRef;
+        }
+
+        this.overlayRef = this.overlay.create({
+            direction: undefined,
+            positionStrategy: this.getPositionStrategy(this.tooltipContentDirective().position()),
+            scrollStrategy: this.overlay.scrollStrategies.close()
+        });
+
+        this.overlayRef
+            .detachments()
+            .pipe(takeUntilDestroyed(this.destroyRef))
+            .subscribe(() => this.detach());
+
+        this.handleOverlayKeydown();
+        this.handlePointerDownOutside();
+
+        return this.overlayRef;
+    }
+
+    private show(): void {
+        this.overlayRef = this.createOverlayRef();
+
+        this.detach();
+
+        this.portal =
+            this.portal ||
+            new TemplatePortal(this.tooltipContentDirective().templateRef, this.viewContainerRef, {
+                state: this.state,
+                side: this.tooltipContentDirective().side
+            });
+
+        this.instance = this.overlayRef.attach(this.portal);
+    }
+
+    private detach(): void {
+        if (this.overlayRef?.hasAttached()) {
+            this.overlayRef.detach();
+        }
+    }
+
+    private hide(): void {
+        if (this.isControlledExternally && this.open()) {
+            return;
+        }
+
+        asyncScheduler.schedule(() => {
+            this.instance?.destroy();
+        }, this.tooltipConfig.hideDelayDuration ?? 0);
+    }
+
+    private getPositionStrategy(connectedPosition: ConnectedPosition): PositionStrategy {
+        return this.overlay
+            .position()
+            .flexibleConnectedTo(this.tooltipTriggerElementRef())
+            .withFlexibleDimensions(false)
+            .withPositions([
+                connectedPosition
+            ])
+            .withLockedPosition();
+    }
+
+    private readonly onIsOpenChangeEffect = effect(() => {
+        const isOpen = this.isOpen();
+
+        untracked(() => {
+            if (isOpen) {
+                this.show();
+            } else {
+                this.hide();
+            }
+        });
```

Comment:
check it – https://angular.dev/guide/signals#effect-cleanup-functions
Ensure all effect subscriptions clean up resources when destroyed to avoid memory leaks.

```
onCleanup(() => this.detach());
```

Reply:
You might need cleanup if:
- The effect attaches listeners or subscriptions.
- The effect creates new resources (like DOM nodes or overlays) that need to be removed.
- The effect schedules tasks (e.g., setTimeout or RxJS streams) that should be canceled when the effect is re-run or destroyed.

Reply:
@pimenovoleg ok, thanks. I'll get to this later